test/controller/confirmation.controller.spec.unit.ts

```diff
@@ -60,4 +64,40 @@ describe("confirmation screen tests", () => {
     expect(response.text).toContain(OBJECTION_ID);
     expect(response.text).toContain(email);
   });
+
+  it("should remove the objection id from the session before showing the confirmation screen", async () => {
+    mockRetrieveUserEmailFromSession.mockReturnValueOnce(email);
+    mockRetrieveFromObjectionSession.mockReturnValueOnce(OBJECTION_ID);
+
+    mockDeleteObjectionSessionValue.mockReset();
+
+    const response = await request(app).get(OBJECTIONS_CONFIRMATION)
+      .set("Referer", "/")
+      .set("Cookie", [`${COOKIE_NAME}=123`]);
+
+    expect(mockDeleteObjectionSessionValue).toHaveBeenCalledTimes(1);
+    expect(mockDeleteObjectionSessionValue).toHaveBeenCalledWith(SESSION, SESSION_OBJECTION_ID);
+
+    expect(response.status).toEqual(200);
+    expect(response.text).toContain(OBJECTION_ID);
+    expect(response.text).toContain(email);
+  });
+
+  it("should land on error screen if no session is available", async () => {
+    mockRetrieveUserEmailFromSession.mockReturnValueOnce(email);
+    mockRetrieveFromObjectionSession.mockReturnValueOnce(OBJECTION_ID);
+
+    mockSessionMiddleware.mockReset();
+    mockSessionMiddleware.mockImplementation((next: NextFunction) => {
```

Comment:
I'd go with mockImplementationOnce here, i'm not sure what order tests will run in.